syne_tune/tuner_callback.py

```diff
@@ -149,37 +152,38 @@ def on_tuning_end(self):
 
 
 class TensorboardCallback(TunerCallback):
+
     def __init__(
             self,
-
-            metric_to_optimize: 'str',
-            mode: 'str',
-            output_metrics: List[str] = [],
+            ignore_metrics: Optional[List[str]] = None,
+            target_metric: Optional[str] = None,
+            mode: Optional[str] = 'min'
     ):
         """
-        Simple callback that logs different statistics such that we can visualize them with Tensorboard.
+        Simple callback that logs metric reported in the train function such that we can visualize with Tensorboard.
 
-        :param metric_to_optimize: The metric that we aim to optimize. We report the cumulative optimum over time. This
-        should be the same metric that we pass to the scheduler / searcher.
-        :param mode: Defines whether we minimize ('min') or maximize ('max'). Should be the same what we
-        pass to scheduler / searcher.
-        :param output_metrics: All other metrics that we want to log. Make sure that these metrics
-         are reported correctly in the train function to Syne Tune via the report(...) function.
+        :param ignore_metrics: Defines which metrics should be ignored. If None, all metrics are reported
+         to Tensorboard.
+        :param target_metric: Defines the metric we aim to optimize. If this argument is set, we report
+        the cumulative optimum of this metric as well as the optimal hyperparameters we have found so far.
+        :param mode: Determined whether we maximize ('max') or minimize ('min') the target metric.
         """
         self.results = []
 
-        self.output_metrics = output_metrics
-        self.mode = mode
-        self.metric = metric_to_optimize
+        self.ignore_metrics = ignore_metrics
 
         self.curr_best_value = None
         self.curr_best_config = None
 
         self._start_time_stamp = None
         self.writer = None
```

Comment:
Can we use `_` for fields consistently in this class? Right now some fields are with `_` and others aren't whereas it seems that all fields would be private.